jules-julesforrest/hw_3/2-json-exercise/app.js

```diff
@@ -0,0 +1,139 @@
+  "someKey": "someValue",
+  "innerArray": [
+    "alpha",
+    "beta",
+    "gamma",
+    "delta"
+  ]
+};
+
+// [ Step 1 ] Use JSONLint to determine if something is valid JSON.
+// Copy the entire object above (including the opening and closing braces {...}, but NOT 'var notActuallyJSON =' or the trailing semicolon).
+// Then go to http://jsonlint.com/ and paste it into the validator. Does it pass the test of being valid JSON?
+
+
+// [ Step 2 ] The following statement is valid JavaScript, but the object it creates is not written in valid JSON syntax.
+// Use JSONLint to validate the object in the following line; then, edit the object so it passes the validator.
+var eddie = {
+  "name":"Eddie Vedder",
+  "age":"49"};
+
+
+// [ Step 3 ] Turn the `grungeAlbums` JavaScript object below into a JSON string, and store it in the variable `grungeAlbumsJSON`.
+// Then, convert `grungeAlbumsJSON` back into a JavaScript object, and compare it to the original `grungeAlbums`. Are they the same?
+
+// Hint: To turn a JS Object into JSON, use the .stringify method  ( JSON.stringify(someObject) )
+// (See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)
+
+var grungeAlbums = {
+  "albums":[
+    {
+        "name": "Bleach",
+        "artist": "Nirvana",
+        "unitsSold": 1700000
+    },
+    {
+        "name": "Nevermind",
+        "artist": "Nirvana",
+        "unitsSold": 30000000
+    },
+    {
+        "name": "In Utero",
+        "artist": "Nirvana",
+        "unitsSold": 15000000
+    },
+    {
+        "name": "Ten",
+        "artist": "Pearl Jam",
+        "unitsSold": 10000000
+    },
+    {
+        "name": "Vs",
+        "artist": "Pearl Jam",
+        "unitsSold": 6100000
+    },
+    {
+        "name": "Vitalogy",
+        "artist": "Pearl Jam",
+        "unitsSold": 4770000
+    }
+  ]
+};
+
+var grungeAlbumsJSON = JSON.stringify({
+  "albums":[
+    {
+        "name": "Bleach",
+        "artist": "Nirvana",
+        "unitsSold": 1700000
+    },
+    {
+        "name": "Nevermind",
+        "artist": "Nirvana",
+        "unitsSold": 30000000
+    },
+    {
+        "name": "In Utero",
+        "artist": "Nirvana",
+        "unitsSold": 15000000
+    },
+    {
+        "name": "Ten",
+        "artist": "Pearl Jam",
+        "unitsSold": 10000000
+    },
+    {
+        "name": "Vs",
+        "artist": "Pearl Jam",
+        "unitsSold": 6100000
+    },
+    {
+        "name": "Vitalogy",
+        "artist": "Pearl Jam",
+        "unitsSold": 4770000
+    }
+  ]
+});
+
+console.log(grungeAlbumsJSON);
+
+// [ Step 4 ] Take `grungeAlbumsJSON` and convert it back into a JavaScript object.
+
+grungeAlbumsJSON = JSON.parse(grungeAlbumsJSON);
+console.log(grungeAlbumsJSON);
+
+// Then, for each album in the list, do a console log printing out the album name, artist, and units sold.
+// Each album should be formatted as follows:
+
+var i;
+
+for (i = 0; i < grungeAlbumsJSON.albums.length; i++) {
```

Comment:
I've noticed a few instances where `i` either goes undeclared or is simply declared outside the forLoop. That's OK, and still logical, but not functionally different than declaring & defining `i` all at once with `var i = 0;`.